### PR TITLE
Fix lwip implementation to use appropriate type for timeouts for socket options setting

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
The API specifies an int of ms but lwip can also use timeval based on build options. This translates the values between implementation and API.

### CODE REVIEW

**Comment by paul-szczepanek-arm** [comment]
To avoid similar problems we should specify the options in the API, see: https://github.com/MDK-Packs/IoT_Socket/issues/8

File: source/lwip/iot_socket.c [hunk 1]
```diff
@@ -611,15 +619,21 @@ int32_t iotSocketSetOpt (int32_t socket, int32_t opt_id, const void *opt_val, ui
       }
       break;
     case IOT_SOCKET_SO_RCVTIMEO:
-      rc = setsockopt(socket, SOL_SOCKET, SO_RCVTIMEO,  (const char *)opt_val, opt_len);
-      if (rc == 0) {
-        sock_attr[socket-LWIP_SOCKET_OFFSET].tv_sec  = *(const uint32_t *)opt_val / 1000U;
-        sock_attr[socket-LWIP_SOCKET_OFFSET].tv_msec = *(const uint32_t *)opt_val % 1000U;
+    case IOT_SOCKET_SO_SNDTIMEO: ; /* do not remove the semi-colon */
```
Comment (0xc0170) [inline source/lwip/iot_socket.c:622]: I would add to the commit message the intention with `;` being placed here. It's important detail why are we doing this.

**Comment by paul-szczepanek-arm** [comment]
I added a comment in the commit message

**Comment by RobertRostohar** [comment]
The commit (3188c55) should properly fix the lwIP implementation. Pleas check and close this PR.

**Comment by RobertRostohar** [comment]
Closing due to being fixed and no further feedback.

**Pull request closed without merging**